web/database/migrations/2021_08_06_003806_create_user_messages_table.php

```diff
@@ -0,0 +1,41 @@
+<?php
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateUserMessagesTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('user_messages', function (Blueprint $table) {
+            $table->uuid('id')->primary();
+            $table->foreignId('from_user_id')->index();
+            $table->foreignId('to_user_id')->nullable()->index();
+            $table->string('subject', 80)->nullable();
+            $table->string('body', 50000);
+            $table->unsignedTinyInteger('flags');
+            $table->timestamps();
+            $table->softDeletes();
```

Comment:
If we have soft deletes on for PMs, will there be a "trash" concept, or is it only soft deletes in the code but there's no practical way for users to recover them? (Or something else)

Reply:
Yes, I didn't bother with helper methods for trashed items because they already exist on the parent class, which oddly enough is called `trashed()`.